core/src/main/java/io/grpc/util/RoundRobinLoadBalancerFactory.java

```diff
@@ -245,6 +271,74 @@ private ConnectivityState getAggregatedState() {
       aCopy.removeAll(b);
       return aCopy;
     }
+
+    /**
+     * Holds stickiness related states: The stickiness key, a registry mapping stickiness values to
+     * the associated Subchannel Ref, and a map from Subchannel to Subchannel Ref.
+     */
+    private static final class StickinessState {
+      static final int MAX_ENTRIES = 100000;
+
+      final Key<String> key;
+      final Map<String, Ref<Subchannel>> stickinessMap =
+          new LinkedHashMap<String, Ref<Subchannel>>() {
+            @Override
+            protected boolean removeEldestEntry(Map.Entry<String,Ref<Subchannel>> eldest) {
+              return size() > MAX_ENTRIES;
+            }
+          };
+
+      final Map<Subchannel, Ref<Subchannel>> subchannelRefs =
+          new HashMap<Subchannel, Ref<Subchannel>>();
+
+      StickinessState(@Nonnull String stickinessKey) {
+        this.key = Key.of(stickinessKey, Metadata.ASCII_STRING_MARSHALLER);
```

Comment:
May want to protect ourselves from the key being `-bin`, since it could be in the future that support is added.